plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -1026,6 +1026,35 @@ fu_genesys_tsdigit_value(gchar c)
 	return g_ascii_digit_value(c);
 }
 
+static gboolean
+fu_genesys_usbhub_device_validate_token(FuGenesysUsbhubDevice *self, GError **error)
+{
+	const gchar *valid_tokens[] = {"GL 3.0 Hub", "GL 3.1 Hub", NULL};
+	g_autofree gchar *token = NULL;
+	g_autoptr(GBytes) token_blob = NULL;
+
+	/* get 0x80 string descriptor */
+	token_blob = g_usb_device_get_string_descriptor_bytes_full(
+	    fu_usb_device_get_dev(FU_USB_DEVICE(self)),
+	    0x80,
+	    G_USB_DEVICE_LANGID_ENGLISH_UNITED_STATES,
+	    64,
+	    error);
+	if (token_blob == NULL)
+		return FALSE;
+	token = fu_utf16_to_utf8_bytes(token_blob, G_LITTLE_ENDIAN, error);
```

Comment:
need to shift 2 bytes; descriptor type & length.

Reply:
Oopa, indeed - sorry about that. Does the 2nd fixup work?

Reply:
You missed me forgetting to type "git push" :) Sorry!